rnote-ui/src/canvaswrapper.rs

```diff
@@ -179,20 +181,30 @@ mod imp {
 
         fn properties() -> &'static [glib::ParamSpec] {
             static PROPERTIES: Lazy<Vec<glib::ParamSpec>> = Lazy::new(|| {
-                vec![glib::ParamSpecBoolean::new(
-                    "show-scrollbars",
-                    "show-scrollbars",
-                    "show-scrollbars",
-                    false,
-                    glib::ParamFlags::READWRITE,
-                )]
+                vec![
+                    glib::ParamSpecBoolean::new(
+                        "show-scrollbars",
+                        "show-scrollbars",
+                        "show-scrollbars",
+                        false,
+                        glib::ParamFlags::READWRITE,
+                    ),
+                    glib::ParamSpecBoolean::new(
+                        "restrict-zoom",
+                        "restrict-zoom",
+                        "restrict-zoom",
+                        false,
+                        glib::ParamFlags::READWRITE,
+                    ),
+                ]
             });
             PROPERTIES.as_ref()
         }
 
         fn property(&self, _id: usize, pspec: &glib::ParamSpec) -> glib::Value {
             match pspec.name() {
                 "show-scrollbars" => self.show_scrollbars.get().to_value(),
+                "restrict-zoom" => self.show_scrollbars.get().to_value(),
```

Comment:
The issue lies here - you forgot to change `self.show_scrollbars` to `self.restrict_zoom`. Because of that, `self.property::<bool>("restrict-zoom")` will always return the current value of the scrollbar visibility option instead.

Reply:
Yes, that was it.